sda-download/api/api.md

```diff
@@ -0,0 +1,161 @@
+# API
+
+
+The Download API service provides functionality for downloading files from the Archive.
+It implements the [Data Out API](https://neic-sda.readthedocs.io/en/latest/dataout/#rest-api-endpoints).
+
+Further, it enables the endpoints `/s3` and `/s3-encrypted`, used for htsget.
+
+The response can be restricted to only contain a given range of a file, and the files can be returned encrypted or decrypted.
+
+All endpoints require an `Authorization` header with an access token in the `Bearer` scheme.
+```
+Authorization: Bearer <token>
+```
+### Authenticated Session
+The client can establish a session to skip time-costly visa validations for further requests. Session is based on the `SESSION_NAME=sda_session_key` (configurable name) cookie returned by the server, which should be returned in later requests.
+
+## Service Description
+
+### Endpoints overview:
+
+**Data out API**:
+
+- `/metadata/datasets`
+- `/metadata/datasets/*dataset`
+- `/files/:fileid`
+
+**htsget**:
+
+- `/s3/<datasetid>/<fileid>`
+- `/s3-encrypted/<datasetid>/<fileid>`
+
+### Data out API
+#### Datasets
+The `/metadata/datasets` endpoint is used to display the list of datasets the given token is authorised to access, that are present in the archive.
+##### Request
+```
+GET /metadata/datasets
+```
+##### Response
+```
+[
+    "dataset_1",
+    "dataset_2"
+]
+```
+#### Files
+##### Request
+Files contained by a dataset are listed using the `datasetName` from `/metadata/datasets`.
+```
+GET /metadata/datasets/{datasetName}/files
+```
+**Scheme Parameter**
+The `?scheme=` query parameter is optional. When a dataset contains a scheme, it may sometimes be problematic with reverse proxies.
```

Comment:
```suggestion
The `?scheme=` query parameter is optional. When a dataset contains a scheme, it may sometimes encounter issues with reverse proxies.
```